packages/apps/spirekey-example/src/spirekey.ts

```diff
@@ -0,0 +1,167 @@
+import { IUnsignedCommand } from '@kadena/client';
+
+export class KadenaSpireKey {
+  private _returnUrl: string;
+  private _spireKeyHostname: string;
+  private _user: IUser | null = null;
+  private _transactions: Record<string, IUnsignedCommand> = {};
+
+  constructor(spireKeyHostname: string, returnUrl: string) {
+    this._spireKeyHostname = spireKeyHostname;
+    this._returnUrl = returnUrl;
+    this._loadFromLocalStorage();
+    this.update(window.location);
+  }
+
+  get isLoggedIn() {
+    return this._user !== null;
+  }
+
+  get user() {
+    return this._user;
+  }
+
+  get transactions() {
+    return this._transactions;
+  }
+
+  update(location: Location) {
+    const searchParams = new URLSearchParams(location.search);
+
+    if (searchParams.has('user')) {
+      const userSearch = searchParams.get('user');
+      if (userSearch && userSearch?.length > 0) {
```

Comment:
`userSearch?.length > 0 `is sufficient to check existence and non empty

Reply:
Yes, but typing doesn't accept that as when `userSearch === null` `userSearch.length = == false` and `false > 0` is not a valid statement